sihl/test/web.ml

```diff
@@ -1,6 +1,118 @@
 open Lwt.Syntax
 open Alcotest_lwt
 
+let choose_routers_without_path_builds_paths _ () =
+  let router =
+    Sihl.Web.(
+      choose
+        [ get "" (fun _ -> Lwt.return @@ Opium.Response.of_plain_text "")
+        ; get "foo" (fun _ -> Lwt.return @@ Opium.Response.of_plain_text "")
+        ; get "bar" (fun _ -> Lwt.return @@ Opium.Response.of_plain_text "")
+        ])
+  in
+  let paths =
+    router |> Sihl.Web.routes_of_router |> List.map (fun (_, path, _) -> path)
+  in
+  Alcotest.(check (list string) "builds paths" [ "/"; "/foo"; "/bar" ] paths);
+  Lwt.return ()
+;;
+
+let choose_routers_with_empty_scope_builds_paths _ () =
+  let router =
+    Sihl.Web.(
+      choose
+        ~scope:""
+        [ get "" (fun _ -> Lwt.return @@ Opium.Response.of_plain_text "")
+        ; get "/" (fun _ -> Lwt.return @@ Opium.Response.of_plain_text "")
```

Comment:
It feels a little weird that you can add a slash with this. Shouldn't you have to create another subroute here which is also `""` to get that slash?

Reply:
You should if you want to use the routers correctly. This is a demonstration of incorrect use where we try to do the right thing.
I am thinking about a `sihl-doctor` package that has a command `sihl doctor` that lists all the things you might be doing wrong. I think this is nicer than `Log.info` or even exceptions.

Reply:
Good idea, let's do it like that.
- [ ] Add docs for `get`
- [ ] Add a comment that `sihl-doctor` checks this

Reply:
@aronerben I just realized that we still have to allow using slashes in case someone doesn't want to use routers and just wants to create a URL `/foo/bar/my/long/path` for some reason. (Maybe a monitoring tool polling a hard coded URL or something like that.

Reply:
I created an issue for `sihl doctor` in general though